docs/language/pipelines.md

```diff
@@ -184,6 +194,29 @@ with
 end;
 ```
 
+the same as state window would read:
+
+```trickle
+use std::time::nanos;
+use tremor::system;
+
+define window fifteen_secs from tumbling
+state
+  null
+script
+  match state of
+      null =>
+        let state = system::nanotime();
```

Comment:
this one should use `tremor::system::ingest_ns()`, no? And for making this example showing all the possible constructs of a state based window it would be cool to add a `script from tick` section too. (This would also make it behave exactly as the tumbling window on time)

Reply:
yes, you're right! good call :)